### PR TITLE
boot-qemu.py: fix ctrl+c in gdb

### PR DESCRIPTION
**Pull request by nickdesaulniers**
Detach QEMU from the process group so that SIGINT isn't also delivered to it.

Ignore SIGINT in python when running GDB subprocess.

Link: https://stackoverflow.com/a/13737455
Link: https://stackoverflow.com/q/30925198
Fixes: https://github.com/ClangBuiltLinux/boot-utils/issues/86

### CODE REVIEW

File: boot-qemu.py [hunk 1]
```diff
@@ -764,13 +764,19 @@ def launch_qemu(cfg):
                 utils.die("Port 1234 is already in use, is QEMU running?")
 
             utils.green("Starting QEMU with GDB connection on port 1234...")
-            with subprocess.Popen(qemu_cmd + ["-s", "-S"]) as qemu_process:
+            qemu_cmd += ["-s", "-S"]
```
Comment (nathanchance) [inline boot-qemu.py:767]: Is this just going to add `-s -S` to the QEMU invocation every time it gets restarted? I think this probably wants to be outside of the loop.

**Review by nathanchance — APPROVED** [review]
Aside from the potential nit, looks good to me! Happy this works :)

File: boot-qemu.py (hunk 1 above)
Comment (nickdesaulniers) [reply boot-qemu.py]: good catch!

**Comment by nickdesaulniers** [comment]
The linter isn't happy.

> boot-qemu.py:768:17: W1509: Using preexec_fn keyword which may be unsafe in the presence of threads (subprocess-popen-preexec-fn)

Looks like I might be able to use `process_group=1` in the [`subprocess.Popen` constructor](https://docs.python.org/3/library/subprocess.html#popen-constructor), but seems it's only available in python 3.11+. (I'm on 3.10.8).

Should we disable W1509?

**Comment by nathanchance** [comment]
> Should we disable W1509?

Yes, either in `actions-workflow` as you did or just adding `# pylint: disable-next=subprocess-popen-preexec-fn` above `subprocess.Popen()` here. Does not matter to me, pick your poison; just make sure everything is green before merging.

**Comment by nathanchance** [comment]
You can drop the `pylint` comment now since you merged https://github.com/ClangBuiltLinux/actions-workflows/commit/20ba4b599aef4d6b0bc2e122c398cf4638d32f72. It looks like there is just one outstanding `yapf` warning, then this should be good to go.

**Pull request merged**